guides/_posts/2014-01-11-ifelse.html

```diff
@@ -36,73 +49,83 @@
+But, any property can be used within an `@if`, `@else` conditional.  So, if you want a layout to be responsive to a section's width:
 
 {% highlight css %}
 
-/* GSS: */
-@if #panel[width] >= 800 {
-  /* ... */
-}
-@else #panel[width] >= 500 {
-  /* ... */
+@if section[width] <= 400 {
+  section {
+    article {
+      :first[top] == ^[top];
+      bottom: == :next[top];
+      left: == ^[left];
+      width: == ^[width];
+      height: == ^[height] / 2;
+    }
+  }
 }
-@else {
-  /* ... */
+@else  {
+  section {
+    article {
+      width: == ^[width] / 2;
+      height: == ^[height];
+      top: == :next[top];
+      right: == :next[left];
+    }
+  }
 }
 
-/* Not possible with Media Queries */
 {% endhighlight %}
 
+When the `section` element `width` goes bellows 400px, the `article` elements will be constrain
+by the `@if` block otherwise by the `@else` block.
+
+### Constraints evaluation in conditions
+In order to determine if some conditions hold, GSS needs to evaluate every constraints outside of those conditions before evaluating if
+they are met, which include constraints appearing after the conditions declaration.
+
+As explained in the [GSS constraints guide](/guide/constraints), the order of constraints declaration can affects the
+resulting solution.
 
+The constraint within the condition of the following example will be added to the solver after the outside constraints. Which
+make sens since the condition is true, you want the constraint within that condition to overcome the other evenly strenghted ones.
 
 {% highlight css %}
 
-@if #panel[width] >= 500 {
-  @horizontal .item;  
-  #button {
-    font-size: == 18;
-  }
-}
-@else {
-  @vertical .item;
-  #button {
-    font-size: == 14;
-  }
+#elm[width] == ::window[width];
+
+@if elm[width] < 400 {
+  #elm[height] == 300;
 }
-{% endhighlight %}
 
+#elm[width] == 300;
+#elm[height] == 500;
 
+{% endhighlight %}
 
-{% highlight css %}
-#box1[width] == 9;
-#box2[width] == 19;
+In this example, if the `width` of the screen is smaller than 400px, the `height` of the element will be constrain to 300px even
+though there's a `height` constraint defined after the condition.
 
-/* with ::this pseudo selector */ 
-.box {
-  @if ::[width] < 10 {
-    height: == 10;
-  }
-  @else {
-    height: == 20;
-  }
-}
-{% endhighlight %}
+### Condition looping
+As describe in the [GSS constraints guide](/guide/constraints), constraints are two ways. You need to be well aware of this fact
+when using conditions. GSS is pretty good at handling cases where a looping could occurs but still, be aware that in some
+complexe conditions nesting you could have some situation where you get in a loop but GSS should get your back the vast majority
+of the time.
 
+For example, the following could have led to a loop, but the GSS cassowary solver will figure out the solution for this set of
+constraints without looping:
```

Comment:
Suggesting we rewrite this as:

> For example, since the conditional below changes a variable that is being checked, it would appear that it will lead to an infinite loop. However, the underlying constraint solver in GSS can determine a solution that satisfies constraints without looping.

(this assumes that this is even true)